### PR TITLE
Bug fix: the payloadbody checker should not modify auth-tokens

### LINKED ISSUE #324: auth tokens are sometimes printed in the network.testing logs, even with no_tokens_in_logs on
Sometimes, the RESTler logs include auth tokens, despite the option no_tokens_in_logs being turned on.

This issue is to track progress on this.

### PR DESCRIPTION
**Pull request by PatGod**
Closes #324 

This PR fixes a bug in the payloadbody checker: it should not modify auth-tokens when substituting uuid-suffixes in the body of a request. (The bug was: auth-tokens were erroneously included in the 'body' part of a request.)

This PR also includes a new option skip_uuid_substitution to skip this part of the payloadbody checker. This new option can be turned on by adding the following in the engine settings.json file:

```
"checkers": {
    "payloadbody": {
        "skip_uuid_substitution" : true
    }
  },
```

Indeed, the uuid_substitution code should be rewritten - a new issue #429 was opened to track this work-item.

Testing: manually with a RESTler job where the payloadbody checker erroneously modified auth-tokens, resulting in tokens being printed in the network.testing logs and REST API requests returning "401 unauthorized" errors.

### CODE REVIEW

File: restler/checkers/payload_body_checker.py [hunk 1]
```diff
@@ -1135,31 +1138,59 @@ def _exec_request_with_new_body(
...
             rendered_data = seq.resolve_dependencies(rendered_data)
 
-            # substitute if there is UUID suffix
-            original_rendered_data = rendered_data
-            uuid4_suffix_dict = self._get_custom_payload_uuid4_suffix()
-            for uuid4_suffix in uuid4_suffix_dict:
-                suffix = uuid4_suffix_dict[uuid4_suffix]
-                len_suffix = len(suffix)
-                # need the query to partition path and body
+            if not self._skip_uuid_substitution:
```
Comment (marina-p) [inline restler/checkers/payload_body_checker.py:1141]: >_skip_uuid_substitution [](http://example.com/codeflow?start=24&length=23)

Could you please add a comment referencing that this setting may be removed when issue #429 is fixed?

**Review by marina-p — APPROVED** [review]
:shipit:

File: restler/checkers/payload_body_checker.py (hunk 1 above)
Comment (PatGod) [reply restler/checkers/payload_body_checker.py:1141]: @marina-p  Well, removing this option later won't be backwards compatible... Is this wise?

(The option is harmless and is off by default.) 

OK?

Comment (PatGod) [reply restler/checkers/payload_body_checker.py:1141]: I just added a note with issue #429 so we remember this point.

**Pull request merged**